### PR TITLE
2024/March 09 ✅ Dynamic routing is on proces  trendings client-compon…

### PR DESCRIPTION
**Pull request by asliddintursunov**
Before: On home page, trending on ilmli and infinite fetch articles were all client-component and used redux-toolkit. 

After: 
1) Trending component changed into server-component but infinite scroll component is still client component, redux toolkit has been removed from both (Trendings, InfiniteScroll) components.
2) Dynamic routing is on process

### CODE REVIEW

File: app/auth/login/page.tsx [hunk 1]
```diff
@@ -6,29 +6,35 @@ import Password from "../components/Password";
...
 
-  const SendData = useCallback(() => {
-    if (regExpResult.username && regExpResult.password) {
-      console.log("Send to server", userData);
-    } else {
-      console.log("Not valid, do not send to server");
+  useEffect(() => {
+    const validationResult = new Set(Object.values(regExpResult));
+    if (username && password) {
+      console.log(validationResult);
```
Comment (muslimchikkk) [inline app/auth/login/page.tsx:20]: To be removed

File: app/auth/register/form/page.tsx [hunk 2]
```diff
@@ -8,27 +8,34 @@ import Email from "../../components/Email";
...
-  });
-  const SendData = useCallback(() => {
-    if (regExpResult.username && regExpResult.email && regExpResult.password) {
-      console.log("Send to server", userData);
-      router.push("/auth/register/interests");
-    } else {
-      console.log("Not valid, do not send to server");
-    }
-  }, [regExpResult]);
+  const [username, setUsername] = useState<string>("");
```
Comment (muslimchikkk) [inline app/auth/register/form/page.tsx:18]: Better default value is null or undefined for inputs

**Review by muslimchikkk — APPROVED** [review]
LGTM!

File: app/auth/login/page.tsx (hunk 1 above)
Comment (asliddintursunov) [reply app/auth/login/page.tsx:20]: Just console.log??

**Pull request merged**